addons/mail/static/src/new/core/messaging_service.js

```diff
@@ -268,6 +268,29 @@ export class Messaging {
                     this.messageService.insert(data);
                     break;
                 }
+                case "mail.message/delete": {
+                    for (const messageId of notif.payload.message_ids) {
+                        const message = this.store.messages[messageId];
```

Comment:
I believe message should be deleted from store and relations (thread, ...) as well.